scripts/open_url_in_instance.py

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env python3
+
+# This script greatly speeds up execution of qutebrowser commands when a
+# qutebrowser instance already exists.  Call it in place of your system's
+# qutebrowser executable.
+
+import hashlib
+import json
+import os
+import os.path
+import socket
+import subprocess
+import sys
+
+version = '1.6.0'
+protocol_version = 1
+
+m = hashlib.md5()
+m.update(os.environ['USER'].encode('utf8'))
+socket_name = 'ipc-' + m.hexdigest()
+
+socket_path = os.path.join(os.environ['XDG_RUNTIME_DIR'], 'qutebrowser', socket_name)
+
+command = {
+    'args' : sys.argv[1:],
+    'target_arg' : 'null',
+    'version' : version,
+    'protocol_version' : protocol_version,
+    'cwd' : os.getcwd()
+}
+
+try:
+    s = socket.socket(socket.AF_UNIX)
+    s.connect(socket_path)
+    s.sendall((json.dumps(command) + '\n').encode('utf8'))
+    s.close()
+except OSError:
+    for exec_path in reversed(os.get_exec_path()):
```

Comment:
This approach is really clever, nice :). I don't think we need a for loop here though, why not just overwrite the path with the reversed path (maybe `":".join(reversed(os.get_exec_path()))`). Since we quit right after this, I don't think we even need to save it.

If the launch fails, we should have some error thrown (with an error code set), not just a silent exit.

Reply:
Thank you.

I would try the first suggestion in a heartbeat (using `os.pathsep`, though 😉), if Python offered `set_exec_path`.  As it turns out, the situation is [hairier](https://github.com/python/cpython/blob/master/Lib/os.py#L610) than simply calling `os.putenv('PATH', ...)`.

I am split regarding the second suggestion--because if that loops fails, that means qutebrowser isn't even installed.  What would be the rationale for accommodating this?  Also, an earlier commit contained an exception catch of similar paranoia, but @The-Compiler advised to remove it.

... Nevertheless, I aim to please.  See 7dcf838da875bcac982332d4760afe5e0d5e4429.

Reply:
Doing `os.environ["PATH"] = ...` seems to work for changing the lookup of `subprocess.run()`.

Reply:
Done.